gossip/client_test.go

```diff
@@ -105,3 +179,36 @@ func TestClientGossip(t *testing.T) {
 		return nil
 	})
 }
+
+// TestClientNodeID verifies a client's gossip request with correct NodeID.
+func TestClientNodeID(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	local, remote, stopper := startFakeServerGossip(t)
+	disconnected := make(chan *client, 1)
+
+	// Use an insecure context. We're talking to unix socket which are not in the certs.
```

Comment:
This comment is not accurate - this test is using TCP instead of a unix socket (so is the other test in this file that you copied this comment from)